x-pack/plugin/rank-rrf/src/yamlRestTest/resources/rest-api-spec/test/linear/10_linear_retriever.yml

```diff
@@ -1333,3 +1333,605 @@ setup:
+  - match: { hits.hits.2._id: "2" }
+  - close_to: { hits.hits.2._score: { value: 0.8, error: 0.01} }
+
+---
+"linear retriever with top-level normalizer and per-retriever override":
+  - requires:
+      cluster_features: [ "linear_retriever.l2_norm", "linear_retriever.top_level_normalizer" ]
+      reason: "Support for L2 normalization and top-level normalizer in linear retriever"
+  - do:
+      search:
+        index: test
+        body:
+          retriever:
+            linear:
+              normalizer: minmax
+              retrievers: [
+                {
+                  retriever: {
+                    standard: {
+                      query: {
+                        bool: {
+                          should: [
+                            { constant_score: { filter: { term: { keyword: { value: "one" } } }, boost: 10.0 } },
+                            { constant_score: { filter: { term: { keyword: { value: "two" } } }, boost: 5.0 } }
+                          ]
+                        }
+                      }
+                    }
+                  },
+                  weight: 1.0,
+                  normalizer: l2_norm
+                },
+                {
+                  retriever: {
+                    knn: {
+                      field: "vector",
+                      query_vector: [4],
+                      k: 2,
+                      num_candidates: 10
+                    }
+                  },
+                  weight: 2.0
+                }
+              ]
+
+  - match: { hits.total.value: 4 }
+  - match: { hits.hits.0._id: "4" }
+  - close_to: { hits.hits.0._score: { value: 2.0, error: 0.01} }
+  - match: { hits.hits.1._id: "1" }
+  - close_to: { hits.hits.1._score: { value: 0.89, error: 0.01} }
+  - match: { hits.hits.2._id: "2" }
+  - close_to: { hits.hits.2._score: { value: 0.45, error: 0.01} }
+
+---
+"linear retriever with top-level normalizer - empty results":
+  - requires:
+      cluster_features: [ "linear_retriever.top_level_normalizer" ]
+      reason: "Support for top-level normalizer in linear retriever"
+  - do:
+      search:
+        index: test
+        body:
+          retriever:
+            linear:
+              normalizer: minmax
+              retrievers: [
+                {
+                  retriever: {
+                    standard: {
+                      query: {
+                        term: {
+                          keyword: {
+                            value: "nonexistent"
+                          }
+                        }
+                      }
+                    }
+                  },
+                  weight: 1.0
+                }
+              ]
+
+  - match: { hits.total.value: 0 }
+  - length: { hits.hits: 0 }
+
+---
+"Linear retriever with per-retriever normalizer override":
+  - requires:
+      cluster_features: [ "linear_retriever.l2_norm", "linear_retriever.top_level_normalizer" ]
```

Comment:
Why do you specify two different cluster features here? That shouldn't be necessary, we should only have to specify `linear_retriever.top_level_normalizer`?